src/Provider/Yandex/Model/YandexAddress.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\Yandex\Model;
+
+use Geocoder\Model\Address;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class YandexAddress extends Address
+{
+    /**
+     * @var string|null
+     */
+    private $precision;
+
+    /**
+     * The name of this location.
+     *
+     * @var string|null
+     */
+    private $name;
+
+    /**
+     * @return null|string
+     */
+    public function getPrecision()
+    {
+        return $this->precision;
+    }
+
+    /**
+     * @param null|string $precision
+     *
+     * @return YandexAddress
+     */
+    public function setPrecision($precision)
```

Comment:
> Compared with:

Why not? Provider makes translation from external API to your generic `Address` class, it is OK to have some dirty code here, isn't it? Offtopic: https://wiki.php.net/rfc/named_params

If you can't avoid some setter-like method, I suggest to use `with`-er syntax:

```php
$address = $address->withFoo('bar'); // with* clones the object and return changed one
```
At least, it makes object immutable.